lld/test/ELF/libsearch.s

```diff
@@ -49,6 +49,9 @@
 // RUN: ld.lld -o %t3 %t.o -L%t.dir -lls
 // RUN: llvm-readobj --symbols %t3 | FileCheck --check-prefix=DYNAMIC %s
 
+// Should use static when dynamic exists in search path
+// RUN: ld.lld --relocatable -o %t3 %t.o -L%t.dir -lls
```

Comment:
We should test the symbol table of output file. Newer tests prefer `llvm-readelf -s` to `llvm-readobj -s`

https://maskray.me/blog/2021-08-08-toolchain-testing#the-test-checks-too-little

Reply:
not sure how use for `llvm-readelf -s`. 

```
llvm-readobj --symbols  test/ELF/Output/libsearch.s.tmp3reloc | FileCheck --check-prefix=STATIC /home/anna/src/llvm-project/lld/test/ELF/libsearch.s
```
seems to work but `llvm-readelf -s` doesn't seem to be compatible with how i'm using FileCheck.

Reply:
i think i figured it out. added the patterns under `RELOCATABLE`.

```
// RUN: ld.lld --relocatable -o %t3 %t.o -L%t.dir -lls
// RUN: llvm-readelf -s -h %t3 | FileCheck --check-prefix=RELOCATABLE %s
// RELOCATABLE: Type: REL (Relocatable file)
// RELOCATABLE: _static
```

<details><summary>llvm-readelf -s -h %t3</summary>

```
ELF Header:
  Magic:   7f 45 4c 46 02 01 01 00 00 00 00 00 00 00 00 00
  Class:                             ELF64
  Data:                              2's complement, little endian
  Version:                           1 (current)
  OS/ABI:                            UNIX - System V
  ABI Version:                       0
  Type:                              REL (Relocatable file)
  Machine:                           Advanced Micro Devices X86-64
  Version:                           0x1
  Entry point address:               0x0
  Start of program headers:          0 (bytes into file)
  Start of section headers:          256 (bytes into file)
  Flags:                             0x0
  Size of this header:               64 (bytes)
  Size of program headers:           0 (bytes)
  Number of program headers:         0
  Size of section headers:           64 (bytes)
  Number of section headers:         6
  Section header string table index: 4

Symbol table '.symtab' contains 5 entries:
   Num:    Value          Size Type    Bind   Vis       Ndx Name
     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT   UND 
     1: 0000000000000000     0 SECTION LOCAL  DEFAULT     1 .text
     2: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT     1 _start
     3: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT     1 _bar
     4: 0000000000000000     0 NOTYPE  GLOBAL DEFAULT     1 _static
```
</details>